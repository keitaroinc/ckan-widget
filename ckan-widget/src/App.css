.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
.dataset-item {
  border: 1px solid #c4c4c4;
  padding: 10px 20px;
  margin: 24px 0;
  display:flex;
  align-items: center;
  justify-content: space-between;
}

.dataset-body {
    margin: 0 20px;
  flex-grow: 1;
  h2 {
    margin-bottom: 0;
  }
}
.dataset-icon{
    max-width: 100%;
    height: auto;
    vertical-align: middle;
    border: 0;
}
.facet-item {
  border: 2px solid #c4c4c4;
  padding: 10px 20px;
  margin: 10px 0;
  display:flex;
  align-items: center;
  justify-content: space-between;
}
.datasets-total{
    float:left;
}
.search-item {
  border: 2px solid #c4c4c4;
  padding: 30px 20px;
  margin: 10px 0;
  display:flex;
  align-items: center;
  justify-content: space-between;
}