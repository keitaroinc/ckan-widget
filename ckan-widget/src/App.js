import React, { Component } from 'react';
import './App.css';
import DatasetSearchBar from './components/container/DatasetSearchBar';
import DatasetInfoList from './components/container/DatasetInfoList';
import FacetList from './components/container/FacetList';
import DatasetSort from './components/container/DatasetSort';
import TotalDatasets from './components/container/TotalDatasets'
import DatasetsPerPage from './components/container/DatasetsPerPage'
import Pagination from './components/container/Pagination'


class App extends Component {
  render() {
    return (
      <div className="App container-fluid">
        <div className="row">
          <div className="col-lg-8">
            <DatasetSearchBar />
          </div>
        </div>
        <div className="row">
          <div className="col-lg-8">
                <TotalDatasets />
                <DatasetsPerPage />
                <DatasetInfoList />
                <Pagination />
          </div>
          <div className="col-lg-4">
              <DatasetSort />
              <FacetList />
          </div>
        </div>
      </div>
    );
  }
}

export default App;
