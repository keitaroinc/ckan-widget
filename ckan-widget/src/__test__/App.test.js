import React from 'react';
import { shallow } from 'enzyme'
import App from '../App';


describe('App', () => {
  it('should be defined', () => {
    expect(App).toBeDefined();
  });

  it('should render correctly', () => {
    var config = {
      ckan_api: 'https://demo.org',
      ckan_organizations: undefined,
      ckan_groups: undefined,
      ckan_tags: undefined,
      ckan_facets: undefined,
      data_sort: 'title_string asc',
      result_page_size: 25,
      thumbnails_display: false
    }

    const component = shallow(
      <App config={config} />
    );

    expect(component).toMatchSnapshot();
  });
});