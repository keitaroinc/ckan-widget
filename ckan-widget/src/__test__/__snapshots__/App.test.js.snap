// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="bg-secondary"
>
          <div
                    className="container-fluid"
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col-lg-10 offset-lg-1"
                              >
                                        <Connect(DatasetSearchBar) />
                              </div>
                    </div>
          </div>
</div>,
        <div
          className="container-fluid my-5"
>
          <div
                    className="row"
          >
                    <div
                              className="col-lg-10 offset-lg-1"
                    >
                              <div
                                        className="row"
                              >
                                        <div
                                                  className="col-lg-8"
                                        >
                                                  <div
                                                            className="d-flex justify-content-between"
                                                  >
                                                            <Connect(TotalDatasets) />
                                                            <Connect(DatasetsPerPage) />
                                                  </div>
                                                  <Connect(DatasetInfoList) />
                                                  <nav
                                                            className="my-5"
                                                  >
                                                            <Connect(Pagination) />
                                                  </nav>
                                        </div>
                                        <div
                                                  className="col-lg-4"
                                        >
                                                  <Connect(DatasetSort) />
                                                  <Connect(FacetList) />
                                        </div>
                              </div>
                    </div>
          </div>
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container-fluid"
>
            <div
                        className="row"
            >
                        <div
                                    className="col-lg-10 offset-lg-1"
                        >
                                    <Connect(DatasetSearchBar) />
                        </div>
            </div>
</div>,
          "className": "bg-secondary",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
>
              <div
                            className="col-lg-10 offset-lg-1"
              >
                            <Connect(DatasetSearchBar) />
              </div>
</div>,
            "className": "container-fluid",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-lg-10 offset-lg-1"
>
                <Connect(DatasetSearchBar) />
</div>,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(DatasetSearchBar) />,
                "className": "col-lg-10 offset-lg-1",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
>
            <div
                        className="col-lg-10 offset-lg-1"
            >
                        <div
                                    className="row"
                        >
                                    <div
                                                className="col-lg-8"
                                    >
                                                <div
                                                            className="d-flex justify-content-between"
                                                >
                                                            <Connect(TotalDatasets) />
                                                            <Connect(DatasetsPerPage) />
                                                </div>
                                                <Connect(DatasetInfoList) />
                                                <nav
                                                            className="my-5"
                                                >
                                                            <Connect(Pagination) />
                                                </nav>
                                    </div>
                                    <div
                                                className="col-lg-4"
                                    >
                                                <Connect(DatasetSort) />
                                                <Connect(FacetList) />
                                    </div>
                        </div>
            </div>
</div>,
          "className": "container-fluid my-5",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-lg-10 offset-lg-1"
>
              <div
                            className="row"
              >
                            <div
                                          className="col-lg-8"
                            >
                                          <div
                                                        className="d-flex justify-content-between"
                                          >
                                                        <Connect(TotalDatasets) />
                                                        <Connect(DatasetsPerPage) />
                                          </div>
                                          <Connect(DatasetInfoList) />
                                          <nav
                                                        className="my-5"
                                          >
                                                        <Connect(Pagination) />
                                          </nav>
                            </div>
                            <div
                                          className="col-lg-4"
                            >
                                          <Connect(DatasetSort) />
                                          <Connect(FacetList) />
                            </div>
              </div>
</div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="row"
>
                <div
                                className="col-lg-8"
                >
                                <div
                                                className="d-flex justify-content-between"
                                >
                                                <Connect(TotalDatasets) />
                                                <Connect(DatasetsPerPage) />
                                </div>
                                <Connect(DatasetInfoList) />
                                <nav
                                                className="my-5"
                                >
                                                <Connect(Pagination) />
                                </nav>
                </div>
                <div
                                className="col-lg-4"
                >
                                <Connect(DatasetSort) />
                                <Connect(FacetList) />
                </div>
</div>,
              "className": "col-lg-10 offset-lg-1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="col-lg-8"
>
                    <div
                                        className="d-flex justify-content-between"
                    >
                                        <Connect(TotalDatasets) />
                                        <Connect(DatasetsPerPage) />
                    </div>
                    <Connect(DatasetInfoList) />
                    <nav
                                        className="my-5"
                    >
                                        <Connect(Pagination) />
                    </nav>
</div>,
                  <div
                    className="col-lg-4"
>
                    <Connect(DatasetSort) />
                    <Connect(FacetList) />
</div>,
                ],
                "className": "row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="d-flex justify-content-between"
>
                        <Connect(TotalDatasets) />
                        <Connect(DatasetsPerPage) />
</div>,
                      <Connect(DatasetInfoList) />,
                      <nav
                        className="my-5"
>
                        <Connect(Pagination) />
</nav>,
                    ],
                    "className": "col-lg-8",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <Connect(TotalDatasets) />,
                          <Connect(DatasetsPerPage) />,
                        ],
                        "className": "d-flex justify-content-between",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {},
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {},
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                      ],
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Connect(Pagination) />,
                        "className": "my-5",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": "nav",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Connect(DatasetSort) />,
                      <Connect(FacetList) />,
                    ],
                    "className": "col-lg-4",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="bg-secondary"
>
            <div
                        className="container-fluid"
            >
                        <div
                                    className="row"
                        >
                                    <div
                                                className="col-lg-10 offset-lg-1"
                                    >
                                                <Connect(DatasetSearchBar) />
                                    </div>
                        </div>
            </div>
</div>,
          <div
            className="container-fluid my-5"
>
            <div
                        className="row"
            >
                        <div
                                    className="col-lg-10 offset-lg-1"
                        >
                                    <div
                                                className="row"
                                    >
                                                <div
                                                            className="col-lg-8"
                                                >
                                                            <div
                                                                        className="d-flex justify-content-between"
                                                            >
                                                                        <Connect(TotalDatasets) />
                                                                        <Connect(DatasetsPerPage) />
                                                            </div>
                                                            <Connect(DatasetInfoList) />
                                                            <nav
                                                                        className="my-5"
                                                            >
                                                                        <Connect(Pagination) />
                                                            </nav>
                                                </div>
                                                <div
                                                            className="col-lg-4"
                                                >
                                                            <Connect(DatasetSort) />
                                                            <Connect(FacetList) />
                                                </div>
                                    </div>
                        </div>
            </div>
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container-fluid"
>
              <div
                            className="row"
              >
                            <div
                                          className="col-lg-10 offset-lg-1"
                            >
                                          <Connect(DatasetSearchBar) />
                            </div>
              </div>
</div>,
            "className": "bg-secondary",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="row"
>
                <div
                                className="col-lg-10 offset-lg-1"
                >
                                <Connect(DatasetSearchBar) />
                </div>
</div>,
              "className": "container-fluid",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="col-lg-10 offset-lg-1"
>
                  <Connect(DatasetSearchBar) />
</div>,
                "className": "row",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(DatasetSearchBar) />,
                  "className": "col-lg-10 offset-lg-1",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
>
              <div
                            className="col-lg-10 offset-lg-1"
              >
                            <div
                                          className="row"
                            >
                                          <div
                                                        className="col-lg-8"
                                          >
                                                        <div
                                                                      className="d-flex justify-content-between"
                                                        >
                                                                      <Connect(TotalDatasets) />
                                                                      <Connect(DatasetsPerPage) />
                                                        </div>
                                                        <Connect(DatasetInfoList) />
                                                        <nav
                                                                      className="my-5"
                                                        >
                                                                      <Connect(Pagination) />
                                                        </nav>
                                          </div>
                                          <div
                                                        className="col-lg-4"
                                          >
                                                        <Connect(DatasetSort) />
                                                        <Connect(FacetList) />
                                          </div>
                            </div>
              </div>
</div>,
            "className": "container-fluid my-5",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-lg-10 offset-lg-1"
>
                <div
                                className="row"
                >
                                <div
                                                className="col-lg-8"
                                >
                                                <div
                                                                className="d-flex justify-content-between"
                                                >
                                                                <Connect(TotalDatasets) />
                                                                <Connect(DatasetsPerPage) />
                                                </div>
                                                <Connect(DatasetInfoList) />
                                                <nav
                                                                className="my-5"
                                                >
                                                                <Connect(Pagination) />
                                                </nav>
                                </div>
                                <div
                                                className="col-lg-4"
                                >
                                                <Connect(DatasetSort) />
                                                <Connect(FacetList) />
                                </div>
                </div>
</div>,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="row"
>
                  <div
                                    className="col-lg-8"
                  >
                                    <div
                                                      className="d-flex justify-content-between"
                                    >
                                                      <Connect(TotalDatasets) />
                                                      <Connect(DatasetsPerPage) />
                                    </div>
                                    <Connect(DatasetInfoList) />
                                    <nav
                                                      className="my-5"
                                    >
                                                      <Connect(Pagination) />
                                    </nav>
                  </div>
                  <div
                                    className="col-lg-4"
                  >
                                    <Connect(DatasetSort) />
                                    <Connect(FacetList) />
                  </div>
</div>,
                "className": "col-lg-10 offset-lg-1",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="col-lg-8"
>
                      <div
                                            className="d-flex justify-content-between"
                      >
                                            <Connect(TotalDatasets) />
                                            <Connect(DatasetsPerPage) />
                      </div>
                      <Connect(DatasetInfoList) />
                      <nav
                                            className="my-5"
                      >
                                            <Connect(Pagination) />
                      </nav>
</div>,
                    <div
                      className="col-lg-4"
>
                      <Connect(DatasetSort) />
                      <Connect(FacetList) />
</div>,
                  ],
                  "className": "row",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <div
                          className="d-flex justify-content-between"
>
                          <Connect(TotalDatasets) />
                          <Connect(DatasetsPerPage) />
</div>,
                        <Connect(DatasetInfoList) />,
                        <nav
                          className="my-5"
>
                          <Connect(Pagination) />
</nav>,
                      ],
                      "className": "col-lg-8",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            <Connect(TotalDatasets) />,
                            <Connect(DatasetsPerPage) />,
                          ],
                          "className": "d-flex justify-content-between",
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "class",
                            "props": Object {},
                            "ref": null,
                            "rendered": null,
                            "type": [Function],
                          },
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "class",
                            "props": Object {},
                            "ref": null,
                            "rendered": null,
                            "type": [Function],
                          },
                        ],
                        "type": "div",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <Connect(Pagination) />,
                          "className": "my-5",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {},
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        "type": "nav",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <Connect(DatasetSort) />,
                        <Connect(FacetList) />,
                      ],
                      "className": "col-lg-4",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
